src/cluster/distributed-state-registry.js

```diff
@@ -180,10 +180,15 @@ module.exports = class DistributedStateRegistry extends EventEmitter {
         nodes: {},
         checkSum: this._createCheckSum(name)
       }
-      this.emit('add', name)
     }
 
+    const exists = this._data[name].nodes[serverName]
+
     this._data[name].nodes[serverName] = true
+
+    if (exists) {
+      this.emit('add', name, serverName)
```

Comment:
UT is not my strength. Feel free to take over the PR... I need this for my listener fork but I don't think mainstream DS really needs it at this point... other than actually fulfilling the "API".